doc/merging.rst

```diff
@@ -0,0 +1,56 @@
+How to merge a branch after approving the PR
+============================================
+
+Requirements
+--------------
+
+- A remote called *upstream* pointing at the central IC repository.
+
+- A remote configured for the author of the PR.
+
+- After the PR is approved, make sure the merge should be done (asking the authors if they didn't get in touch with you).
+
+
+Steps
+-------
+
+#. Fetch *upstream/master*.
+
+#. Reset or rebase your master branch to *upstream/master*.
+
+#. Fetch the branch of the approved PR.
+
+#. Create and check out a local branch on top of the PR branch.
+
+#. Make sure the branch of the approved PR is rebased onto *upstream/master*. If not:
+
+   * Rebase the branch onto *upstream/master*. If there are conflicts, ask the author to resolve them, unless they are obvious.
+
+   * Push to the branch of the PR and wait until the tests finish.
+
+#. Checkout your local *master*.
+
+#. Merge the PR branch into your local *master*, making sure that the merge commit conforms to our requirements. Here are the steps needed to make the merge happen:
+
+   * Disallow fast forward merging: we want an explicit merge commit for each PR.
+
+   * Edit the commit message as described above so that it has the following format:
+
+   <PR number>  <PR title>
+
+   <PR url>
+
+   [author: <author's id>]
+
+   <PR description>  (This is usually the whole first comment in the PR)
+
+   [reviewer: <approver's id>]
+
+   <Reviewer comment> (comment the reviewer left on GitHub)
+
+
+   * Ensure that Git records the reviewer of the PR as the author of the merge commit.
```

Comment:
Hmm, this is probably the most obscure git operation anyone on this project will ever do ... almost certainly the most obscure git operation that a budding NEXT merge-master has ever come across. As such, it could do with some explanation ... on the other hand, explaining it in text rather than interactively con peras seems hideously difficult, especially given the GitHub quirks where it sometimes only recognizes the author's id and sometimes only the full name, or email or whatever. Not sure what to do about this.

Reply:
I don't understand what do you doubt here. To skip this step?

Reply:
Leave it as it is, other than adding a link to the detailed instructions how to achieve this, in the details section we agreed to add.

Reply:
Does this mean 'set the reviewer as the author of merge commit'?

Reply:
Yes, to avoid that the merger person "receives the credits"

Reply:
Not so much about credits, as the fact that who reviewed and accepted it being more interesting and relevant information than who went through the mechanics of merging.